pkg/controllers/cluster/cluster_controller.go

```diff
@@ -511,3 +533,28 @@ func (c *Controller) processTaintBaseEviction(ctx context.Context, cluster *clus
 	}
 	return nil
 }
+
+func (c *Controller) taintClusterByCondition(ctx context.Context, cluster *clusterv1alpha1.Cluster) error {
+	currentReadyCondition := meta.FindStatusCondition(cluster.Status.Conditions, clusterv1alpha1.ClusterConditionReady)
+
+	if currentReadyCondition != nil {
```

Comment:
probably the last question.
Why skip the `nil` condition here?

Reply:
Ready condition nil represent that cluster_status_controller never update cluster status. so we can't determine how to deal with the taint.
Other operation like eviction also has this judgement too.

Reply:
> Other operation like eviction also has this judgement too.

Share a link?

Reply:
https://github.com/karmada-io/karmada/blob/a42c819c101e850ec89f15f8fd7cb8f2b9bf94e9/pkg/controllers/cluster/cluster_controller.go#L367-L373

Reply:
Is this ok?

Reply:
Sorry I missed this. 
I'm thinking if we should add a taint to the cluster when it is registering. cc @Garrybest

Reply:
Should we? It seems that K8s does not add a taint when a node is registering.

Reply:
cc @prodanlabs @pidb who heavily uses the scheduler.